mesonbuild/modules/python.py

```diff
@@ -443,10 +443,8 @@ def find_installation(self, state: 'ModuleState', args: T.Tuple[T.Optional[str]]
             mlog.log('Program', name_or_path or 'python', 'found:', mlog.red('NO'), '(disabled by:', mlog.bold(feature), ')')
             return NonExistingExternalProgram()
 
-        python = self.installations.get(name_or_path)
-        if not python:
-            python = self._find_installation_impl(state, display_name, name_or_path, required)
-            self.installations[name_or_path] = python
+        python = self._find_installation_impl(state, display_name, name_or_path, required)
+        self.installations[name_or_path] = python
```

Comment:
In terms of the actual caching itself, this would be self-defeating. The original purpose of adding an `_impl` method and doing a cache lookup was in commit c5c02b72e132a03ae0284bdd5e15d1675301a37d which explains that it is desirable to avoid doing sanity checks again and again. Here you're switching back to discarding the entire installation every time, so we might as well not look it up at all and in `_get_install_scripts` (the only other place that self.installations is ever used) we can just look up a dict of versions.

Reply:
I'd have thought you'd generally only run `find_installation()` once, but I'm open to suggestions. Should the cache key on the given name rather than the value from the machine file?

Reply:
Okay, I'll try caching against a tuple of both the requested and the resolved name/path.